src/providers/tfc/planProvider.ts

```diff
@@ -0,0 +1,424 @@
+      }
+      if (labels.length > 0) {
+        label = `Planned changes: ${labels.join(', ')}`;
+      }
+    }
+    super(label, vscode.TreeItemCollapsibleState.Expanded);
+  }
+
+  getChildren(): vscode.TreeItem[] {
+    const items: vscode.TreeItem[] = [];
+    this.plannedChanges.forEach((change) => {
+      items.push(new PlannedChangeItem(change));
+    });
+    return items;
+  }
+}
+
+class PlannedChangeItem extends vscode.TreeItem {
+  constructor(public change: Change) {
+    let label = change.resource.addr;
+    if (change.previous_resource) {
+      label = `${change.previous_resource.addr} → ${change.resource.addr}`;
+    }
+
+    super(label, vscode.TreeItemCollapsibleState.None);
+    this.id = change.action + '/' + change.resource.addr;
+    const icon = GetChangeActionIcon(change.action);
+    this.iconPath = icon;
+    this.description = change.action;
+
+    const tooltip: vscode.MarkdownString = new vscode.MarkdownString();
+    if (change.previous_resource) {
+      tooltip.appendMarkdown(
+        `\`${change.previous_resource.addr}\` planned to _${change.action}_ to \`${change.resource.addr}\``,
+      );
+    } else if (change.importing) {
+      tooltip.appendMarkdown(
+        `Planned to _${change.action}_ \`${change.resource.addr}\` (id=\`${change.importing.id}\`)`,
+      );
+    } else {
+      tooltip.appendMarkdown(`Planned to _${change.action}_ \`${change.resource.addr}\``);
+    }
+    this.tooltip = tooltip;
+  }
+}
+
+class DriftChangesItem extends vscode.TreeItem implements ItemWithChildren {
+  constructor(private driftChanges: Change[], summary?: DriftSummary) {
+    let label = `Drifted resources`;
+    if (summary) {
+      const details = [];
+      if (summary.changed > 0) {
+        details.push(`${summary.changed} changed`);
+      }
+      if (summary.deleted > 0) {
+        details.push(`${summary.deleted} deleted`);
+      }
+      label = `Drifted resources: ${details.join(', ')}`;
+    }
+
+    super(label, vscode.TreeItemCollapsibleState.Expanded);
+  }
+
+  getChildren(): vscode.TreeItem[] {
+    const items: vscode.TreeItem[] = [];
+    this.driftChanges.forEach((change) => {
+      items.push(new DriftChangeItem(change));
+    });
+    return items;
+  }
+}
+
+class DriftChangeItem extends vscode.TreeItem {
+  constructor(public change: Change) {
+    let label = change.resource.addr;
+    if (change.previous_resource) {
+      label = `${change.previous_resource.addr} → ${change.resource.addr}`;
+    }
+
+    super(label, vscode.TreeItemCollapsibleState.None);
+    this.id = 'drift/' + change.action + '/' + change.resource.addr;
+    const icon = GetChangeActionIcon(change.action);
+    this.iconPath = icon;
+    const message = GetDriftChangeActionMessage(change.action);
+    this.description = message;
+
+    const tooltip: vscode.MarkdownString = new vscode.MarkdownString();
+    tooltip.appendMarkdown(`\`${change.resource.addr}\` _${message}_`);
+    this.tooltip = tooltip;
```

Comment:
```suggestion
    this.tooltip = new vscode.MarkdownString(`\`${change.resource.addr}\` _${this.description}_`);
```
You can provide an initial value to `MarkdownString()`, so we don't have to call append on an empty string.